pennylane/math/single_dispatch.py

```diff
@@ -128,10 +128,38 @@ def _permutation_parity(perm):
     return parity
 
 
+def _sparse_matrix_power_bruteforce(A, n):
+    """
+    Compute the power of a sparse matrix using brute-force matrix multiplication.
+    Supports only non-negative integer exponents.
+
+    Parameters:
+    A : scipy.sparse matrix
+        The sparse matrix to be exponentiated.
+    n : int
+        The exponent (must be non-negative).
+
+    Returns:
+    scipy.sparse matrix
+        The matrix A raised to the power n.
+    """
+    if n < 0:

```

Comment:
Can we check the scipy version and dispatch to the native implementation if it is supported?

Reply:
https://github.com/PennyLaneAI/pennylane/pull/7055/commits/9784e02b65dbd9df9f4311a18a7dca8d82aa6e82